src/osd/scrubber/scrub_machine.cc

```diff
@@ -1149,11 +1020,25 @@ sc::result ReplicaActiveOp::react(const StartReplica&)
   return transit<ReplicaActiveOp>();
 }
 
+
+sc::result ReplicaActiveOp::react(const ReplicaReserveReq& ev)
+{
+  DECLARE_LOCALS;  // 'scrbr' & 'pg_id' aliases
+  dout(10) << "ReplicaActiveOp::react(const ReplicaReserveReq&)" << dendl;
+  scrbr->get_clog()->warn() << fmt::format(
+      "osd.{} pg[{}]: reservation requested while processing a chunk",
+      scrbr->get_whoami(), scrbr->get_spgid());
+  // exit-n-renter ReplicaActive (aborting the chunk & freeing the reservation
+  // in the process)
+  post_event(ev);
+  return transit<ReplicaActive>();
+}
+
+
 sc::result ReplicaActiveOp::react(const ReplicaRelease& ev)
 {
   dout(10) << "ReplicaActiveOp::react(const ReplicaRelease&)" << dendl;
-  post_event(ev);
-  return transit<sc::shallow_history<ReplicaReserved>>();
+  return transit<ReplicaActive>();
```

Comment:
By contrast, this one is supposed to happen and should result in a reset.

Reply:
That's what the transit "back" to ReplicaActive should cause. Validating this now, as otherwise - this is not the only problem in this PR.

Reply:
Oh, I meant that this seems correct.